test/correctness.cpp

```diff
@@ -0,0 +1,422 @@
+#include <gtest/gtest.h>
+#include <vector>
+#include <format>
+
+#include <piwcsprwmodel.h>
+
+#include "../src/nodetypeinfo.h"
+#include "../src/correctness.h"
+
+using namespace piwcs::prw;
+
+using SType = Section::AllowedTravel;
+
+struct SectionTestInfo{
+    Section::AllowedTravel type;
+    bool inward = false; // Section is directed into node if `true`
+};
+
+template<std::size_t N>
+bool localCorrectnessMinimal(
+        NodeType type,
+        const SectionTestInfo (& sections)[N]
+        ){
+
+    assert(
```

Comment:
Use `_ASSERT` from `src/debug.h`, clang-tidy won't be happy otherwise

Reply:
Adressed in a1b3961.